rllib/core/rl_module/torch/torch_rl_module.py

```diff
@@ -43,6 +44,25 @@ def __init__(self, *args, **kwargs) -> None:
         nn.Module.__init__(self)
         RLModule.__init__(self, *args, **kwargs)
 
+        # If an inference-only class AND self.config.inference_only is True,
+        # remove all attributes that are returned by
+        # `self.get_non_inference_attributes()`.
+        if self.config.inference_only and isinstance(self, InferenceOnlyAPI):
+            for attr in self.get_non_inference_attributes():
+                parts = attr.split(".")
+                if not hasattr(self, parts[0]):
+                    continue
+                target = getattr(self, parts[0])
+                # Traverse from the next part on (if nested).
+                for part in parts[1:]:
```

Comment:
This stops after the first nesting of modules. Can't we just use a recursive implementation that calls itself again? 

Also, this first builds all the components (could be many) and then deletes them again. This could induce some initialization cost which hurts performance in cases where workers need to be rebuild often.

Reply:
This is true, BUT users can always decide to not even build these components in the first place inside their `setup()`. In this case, the above functionality will simply ignore the (already) missing attributes (which it would have deleted otherwise).